### PR TITLE
[spaceship] add client optional parameter in model helpers

### PR DESCRIPTION
**Pull request by bolom**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
https://github.com/fastlane/fastlane/issues/13962
allow to pass client parameter

### CODE REVIEW

File: fake/derived/path/Logs/Package/LogStoreManifest.plist [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>logFormatVersion</key>
+	<integer>10</integer>
+	<key>logs</key>
+	<dict/>
+</dict>
+</plist>
```
Comment (joshdholtz) [inline fake/derived/path/Logs/Package/LogStoreManifest.plist:10]: Oops, looks like this file got added and shouldn't have 🤷‍♂️

File: fake/derived/path/Logs/Test/LogStoreManifest.plist [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>logFormatVersion</key>
+	<integer>10</integer>
+	<key>logs</key>
+	<dict/>
+</dict>
+</plist>
```
Comment (joshdholtz) [inline fake/derived/path/Logs/Test/LogStoreManifest.plist:10]: Same here 🤷‍♂️ I should look into where these are coming from and why they weren't cleaned up

File: fastlane/lib/fastlane/helper/s3_client_helper.rb [hunk 3]
```diff
@@ -46,7 +46,7 @@ def delete_file(bucket_name, file_name)
       end
 
       def find_bucket!(bucket_name)
-        bucket = Aws::S3::Bucket.new(bucket_name, client: client)
+        bucket = Aws::S3::Bucket.new(bucket_name, client: nil)
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/helper/s3_client_helper.rb:49]: Was this an intended part fo this change? 🤔

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Just some small things!

File: fastlane/lib/fastlane/helper/s3_client_helper.rb (hunk 3 above)
Comment (bolom) [reply fastlane/lib/fastlane/helper/s3_client_helper.rb:49]: @joshdholtz craps. wrong copy and past.

**Comment by joshdholtz** [comment]
@googlebot I consent.

File: fake/derived/path/Logs/Test/LogStoreManifest.plist (hunk 2 above)
Comment (rogerluan) [reply fake/derived/path/Logs/Test/LogStoreManifest.plist:10]: Same happens here: these are also not cleaned up when I run tests 😅

File: fastlane/lib/fastlane/helper/s3_client_helper.rb (hunk 3 above)
Comment (joshdholtz) [reply fastlane/lib/fastlane/helper/s3_client_helper.rb:49]: @bolom Do you have a changes that you are still looking to push to fix this? 😊

Comment (bolom) [reply fastlane/lib/fastlane/helper/s3_client_helper.rb:49]: I will do it today. @joshdholtz

Comment (joshdholtz) [reply fastlane/lib/fastlane/helper/s3_client_helper.rb:49]: Thank you!! No rush on my part but just wanted to make sure we could get this merged for you at some point 😁

File: fake/derived/path/Logs/Test/LogStoreManifest.plist (hunk 2 above)
Comment (rogerluan) [reply fake/derived/path/Logs/Test/LogStoreManifest.plist:10]: Thanks, Josh! 🤗 https://github.com/fastlane/fastlane/pull/17559

**Comment by joshdholtz** [comment]
@googlebot I consent.

File: spaceship/lib/spaceship/connect_api/models/app_preview.rb [hunk 4]
```diff
@@ -83,14 +85,14 @@ def self.create(app_preview_set_id: nil, path: nil, wait_for_processing: true, f
...
-          preview = Spaceship::ConnectAPI.patch_app_preview(
+          preview = client.patch_app_preview(
             app_preview_id: preview.id,
             attributes: patch_attributes
           ).first
         rescue => error
           puts("Failed to patch app preview. Update may have gone through so verifying") if Spaceship::Globals.verbose?
 
-          preview = Spaceship::ConnectAPI::AppPreview.get(app_preview_id: preview.id)
+          preview = Spaceship::ConnectAPI::AppPreview.get(client: nil, app_preview_id: preview.id)
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/models/app_preview.rb:95]:
```suggestion
          preview = Spaceship::ConnectAPI::AppPreview.get(client: client, app_preview_id: preview.id)
```

File: spaceship/lib/spaceship/connect_api/models/app_preview.rb [hunk 5]
```diff
@@ -111,20 +113,22 @@ def self.create(app_preview_set_id: nil, path: nil, wait_for_processing: true, f
             puts("Waiting #{sleep_time} seconds before checking status of processing...") if Spaceship::Globals.verbose?
             sleep(sleep_time)
 
-            preview = Spaceship::ConnectAPI::AppPreview.get(app_preview_id: preview.id)
+            preview = Spaceship::ConnectAPI::AppPreview.get(client: nil, app_preview_id: preview.id)
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/models/app_preview.rb:116]:
```suggestion
            preview = Spaceship::ConnectAPI::AppPreview.get(client: client, app_preview_id: preview.id)
```

File: spaceship/lib/spaceship/connect_api/models/app_screenshot.rb [hunk 6]
```diff
@@ -117,7 +118,7 @@ def self.create(app_screenshot_set_id: nil, path: nil, wait_for_processing: true
             sleep(30)
 
             screenshots = Spaceship::ConnectAPI::AppScreenshotSet
-                          .get(app_screenshot_set_id: app_screenshot_set_id)
+                          .get(client: nil, app_screenshot_set_id: app_screenshot_set_id)
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/models/app_screenshot.rb:121]:
```suggestion
                          .get(client: client, app_screenshot_set_id: app_screenshot_set_id)
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
I think I found 3 spots that were missing client other than that I think we are good here!

**Review by joshdholtz — APPROVED** [review]
This is 🔥 I reviewed it two more times tonight and found two more small fixes! But I think we should be good now 😊

**Pull request merged**